src/Morphir/IR/Value.elm

```diff
@@ -76,25 +75,25 @@ import String
 
 {-| Type that represents a value.
 -}
-type Value extra
-    = Literal Literal extra
-    | Constructor FQName extra
-    | Tuple (List (Value extra)) extra
-    | List (List (Value extra)) extra
-    | Record (List ( Name, Value extra )) extra
-    | Variable Name extra
-    | Reference FQName extra
-    | Field (Value extra) Name extra
-    | FieldFunction Name extra
-    | Apply (Value extra) (Value extra) extra
-    | Lambda (Pattern extra) (Value extra) extra
-    | LetDefinition Name (Definition extra) (Value extra) extra
-    | LetRecursion (List ( Name, Definition extra )) (Value extra) extra
-    | Destructure (Pattern extra) (Value extra) (Value extra) extra
-    | IfThenElse (Value extra) (Value extra) (Value extra) extra
-    | PatternMatch (Value extra) (List ( Pattern extra, Value extra )) extra
-    | UpdateRecord (Value extra) (List ( Name, Value extra )) extra
-    | Unit extra
+type Value a
+    = Literal a Literal
+    | Constructor a FQName
+    | Tuple a (List (Value a))
+    | List a (List (Value a))
+    | Record a (List ( Name, Value a ))
+    | Variable a Name
+    | Reference a FQName
+    | Field a (Value a) Name
+    | FieldFunction a Name
+    | Apply a (Value a) (Value a)
+    | Lambda a (Pattern a) (Value a)
+    | LetDefinition a Name (Definition a) (Value a)
+    | LetRecursion a (List ( Name, Definition a )) (Value a)
+    | Destructure a (Pattern a) (Value a) (Value a)
```

Comment:
Why does this have 2 `Value a` args?

Reply:
This should probably be called `DestructureIn` to make it clear that the second value is the `in`